modules/engines/engine-xsjs/src/test/java/com/sap/xsk/synchronizer/ExportGenerationTest.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.synchronizer;
+
+import com.sap.xsk.engine.XSKJavascriptEngineExecutor;
+import com.sap.xsk.exceptions.XSJSLibArtefactCleanerSQLException;
+import com.sap.xsk.exceptions.XSJSLibExportsGenerationSourceNotFoundException;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.eclipse.dirigible.commons.api.scripting.ScriptingException;
+import org.eclipse.dirigible.commons.config.Configuration;
+import org.eclipse.dirigible.commons.config.StaticObjects;
+import org.eclipse.dirigible.core.scheduler.api.ISchedulerCoreService;
+import org.eclipse.dirigible.core.scheduler.service.definition.JobDefinition;
+import org.eclipse.dirigible.core.test.AbstractDirigibleTest;
+import org.eclipse.dirigible.engine.js.graalvm.processor.GraalVMJavascriptEngineExecutor;
+import org.eclipse.dirigible.repository.api.IEntity;
+import org.eclipse.dirigible.repository.api.IRepository;
+import org.eclipse.dirigible.repository.local.LocalRepository;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnitParamsRunner.class)
+public class ExportGenerationTest extends AbstractDirigibleTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(ExportGenerationTest.class);
+
+  public ExportGenerationTest() {
+    // should be executed before parent @Before method as parent would otherwise initialize the DB in a persistent way
+    Configuration.set("DIRIGIBLE_DATABASE_H2_URL", "jdbc:h2:mem:xsk-tests");
+  }
+
+  @Before
+  public void beforeTest() {
+    setUpRepository();
+    cleanup();
```

Comment:
If you executed a previous build with tests and you run a new one right after - the repository points to the folder of the previous run thus the first executing test would not see a clean repository, but the others after it will (due to cleanup on afterTest). Just a precaution?